### PR TITLE
Resolve the issues reported in #110 with bad reporting of errors

### LINKED ISSUE #110: Spectrum 1.1.0 displays test case nested structure wrong in Intellij Idea
Unfortunately version 1.1.0 seems to have introduced funky behavior in Intellij Idea test output structure in case of exception inside tests.

First the test code that throws exception:
[](https://postimg.org/image/d8qpugfsp/)
Second how Intellij Idea displays it with Spectrum 1.0.0
[](https://postimg.org/image/xiy9w8iy5/)
Third the same test with Spectrum 1.1.0
[](https://postimg.org/image/jh43duch1/)

Happens at least with Intellij Idea 2017.1 and 2016.3.5

### PR DESCRIPTION
**Pull request by ashleyfrieze**
Fixes #110 

Errors were reported in the wrong order confusing IDE JUnit runners

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -16,7 +16,7 @@ allprojects {
 dependencies {
   compile group: 'junit', name: 'junit', version: '4.12'
   testCompile group: 'org.hamcrest', name: 'hamcrest-library', version: '1.3'
-  testCompile group: 'org.mockito', name: 'mockito-core', version: '1.10.19'
+  testCompile group: 'org.mockito', name: 'mockito-core', version: '2.7.22'
```
Comment (ashleyfrieze) [inline build.gradle]: Upgraded to the latest `Mockito` to allow easier `argThat`

File: src/main/java/com/greghaskins/spectrum/internal/CompositeTest.java [hunk 2]
```diff
@@ -26,18 +26,15 @@ public boolean isAtomic() {
...
-    runNotifier.addListener(listener);
-    try {
-      for (Child child : suite.children) {
-        if (listener.hasFailedYet()) {
-          child.ignore();
-        }
-        suite.runChild(child, runNotifier);
+  private static void abortOnFailureChildRunner(final Suite suite,
+      final RunReporting<Description, Failure> reporting) {
+    FailureDetectingRunDecorator<Description, Failure> decoratedReporting =
```
Comment (ashleyfrieze) [inline src/main/java/com/greghaskins/spectrum/internal/CompositeTest.java]: This decorator simplifies the method a lot. We no longer add a listener to the `RunNotifier` but instead wrap the `Reporting` with a sentinel.

File: src/main/java/com/greghaskins/spectrum/internal/Suite.java [hunk 3]
```diff
@@ -263,8 +290,6 @@ public boolean isEffectivelyIgnored() {
 
   private boolean hasANonIgnoredChild() {
     return this.children.stream()
-        .filter(child -> !child.isEffectivelyIgnored())
-        .findFirst()
-        .isPresent();
+        .anyMatch(child -> !child.isEffectivelyIgnored());
```
Comment (ashleyfrieze) [inline src/main/java/com/greghaskins/spectrum/internal/Suite.java]: Simplification suggested by IntelliJ.

**Comment by greghaskins** [comment]
@ashleyfrieze looks great to me. If you confirmed that it gets the right IDE behavior, I'm ready to merge. 👍

Also has nice side effect, as you mentioned, of starting to isolate us incrementally away from the JUnit4 `RunNotifier` API.

**Comment by ashleyfrieze** [comment]
I'm pretty happy with this. Can you release it too? I feel bad that the general release has this howler in it?

**Pull request merged**